cJSON.c

```diff
@@ -1871,22 +1871,34 @@ static cJSON_bool add_item_to_array(cJSON *array, cJSON *item)
     }
 
     child = array->child;
-
+    /*
+     * To find the last item int array quickly, we use prev in array
+     */
     if (child == NULL)
     {
         /* list is empty, start new one */
         array->child = item;
+        item->prev = item;
```

Comment:
shouldn't the prev of item be NULL? and the default value of item's prev and next item is NULL, so there is no need to point it, what do you think?

Reply:
if the object is array,  we could use the prev pointer, so we can do array_add more quick by "prev".
we don't need to find the last item one by one

Reply:
if the size of array is large , we could do array_add better

Reply:
oh, I just catch your idea, prev way is indeed faster for large array.

Reply:
yayaya, i use it in my project, the performance improve about 30 percent when create a cjson array which  has thousand's item.
 it could be better, if the element is easier.